### PR TITLE
HCAP-776 | Load testing

### PR DESCRIPTION
**Pull request by arranfw**
https://freshworks.atlassian.net/browse/HCAP-776

- Use K6 to simulate the requests that occur when a user logs in (get user object, get related PEOIs)
- Makefile target
- Requires KC participant user
- All environment variables are documented in load/README.md

Testing so far indicates application crashes when exceeding 100 rps sustained for over a minute. Current values are 75 rps for 2 minutes and low/no failures typically occur.

@vyasworks @fwpushan 
Could you review the work and verify the process is sound before I open it up to the team?

Sample Output for `rate=75` `duration=120`

User Login Sample:
```
scenarios: (100.00%) 1 scenario, 75 max VUs, 2m30s max duration (incl. graceful stop):
  * all: 75.00 iterations/s for 2m0s (maxVUs: 0-75, gracefulStop: 30s)

SUCCESS_REQS...................: 3964   31.951328/s
data_received..................: 17 MB  141 kB/s
data_sent......................: 8.1 MB 66 kB/s
dropped_iterations.............: 7019   56.575775/s
http_req_blocked...............: avg=10.88ms min=1.2µs    med=2.7µs   max=202.8ms  p(90)=58.68ms p(95)=62.54ms 
http_req_connecting............: avg=4.71ms  min=0s       med=0s      max=50.4ms   p(90)=26.16ms p(95)=28.06ms 
http_req_duration..............: avg=1.51s   min=36.44ms  med=1.86s   max=4.62s    p(90)=2.79s   p(95)=3.02s   
  { expected_response:true }...: avg=1.51s   min=36.44ms  med=1.86s   max=4.62s    p(90)=2.79s   p(95)=3.02s   
http_req_failed................: 0.00%  ✓ 0         ✗ 5946
http_req_receiving.............: avg=81.78µs min=18.7µs   med=77.6µs  max=1.15ms   p(90)=121.4µs p(95)=137.97µs
http_req_sending...............: avg=27.6µs  min=6.9µs    med=31.35µs max=309.2µs  p(90)=45.2µs  p(95)=55.9µs  
http_req_tls_handshaking.......: avg=6.11ms  min=0s       med=0s      max=175.11ms p(90)=31.94ms p(95)=34.8ms  
http_req_waiting...............: avg=1.51s   min=36.3ms   med=1.86s   max=4.62s    p(90)=2.79s   p(95)=3.02s   
http_reqs......................: 5946   47.926992/s
iteration_duration.............: avg=4.57s   min=494.87ms med=4.59s   max=6.76s    p(90)=5.29s   p(95)=5.49s   
iterations.....................: 1982   15.975664/s
vus............................: 75     min=36      max=75
vus_max........................: 75     min=36      max=75
```

Static FE Sample:
```
SUCCESS_REQS...................: 13890  431.849035/s
data_received..................: 2.9 GB 92 MB/s
data_sent......................: 3.0 MB 93 kB/s
dropped_iterations.............: 3223   100.205143/s
http_req_blocked...............: avg=6.68ms   min=1µs      med=3.1µs    max=1.25s   p(90)=6.6µs   p(95)=37.9µs  
http_req_connecting............: avg=3.39ms   min=0s       med=0s       max=1.15s   p(90)=0s      p(95)=0s      
http_req_duration..............: avg=418.7ms  min=21.24ms  med=58.33ms  max=6.27s   p(90)=1.71s   p(95)=2.26s   
  { expected_response:true }...: avg=418.7ms  min=21.24ms  med=58.33ms  max=6.27s   p(90)=1.71s   p(95)=2.26s   
http_req_failed................: 0.00%  ✓ 0          ✗ 13890
http_req_receiving.............: avg=363.91ms min=19.9µs   med=131.45µs max=6.18s   p(90)=1.65s   p(95)=2.19s   
http_req_sending...............: avg=35.95µs  min=5.9µs    med=15.8µs   max=5.68ms  p(90)=57.51µs p(95)=86µs    
http_req_tls_handshaking.......: avg=3.27ms   min=0s       med=0s       max=842.8ms p(90)=0s      p(95)=0s      
http_req_waiting...............: avg=54.75ms  min=19.9ms   med=43.5ms   max=2.05s   p(90)=75.18ms p(95)=102.41ms
http_reqs......................: 13890  431.849035/s
iteration_duration.............: avg=2.12s    min=219.69ms med=2.01s    max=6.77s   p(90)=3.3s    p(95)=3.87s   
iterations.....................: 2778   86.369807/s
vus............................: 200    min=88       max=200
vus_max........................: 200    min=88       max=200
```

### CODE REVIEW

File: load/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+## Quick Start
+1. Ensure valid environment variables exist for keycloak:
+   - `DURATION` (Length of load test)
+   - `RATE` (Function calls per second)
+   - `LOAD_KC_AUTH_USERNAME` (Username for KC participant account)
+   - `LOAD_KC_AUTH_PASSWORD` (Password for KC participant account)
+   - `LOAD_KC_AUTH_CLIENTID`
+   - `OS_NAMESPACE_SUFFIX` (dev/test)
+   - `KEYCLOAK_REALM`
+2. Run `make loadtest script=participantLogin.js rate=75 duration=120`
```
Comment (vyasworks) [inline load/README.md:12]: are these two required as ENV vars? looks like docker sets them from make file command.

**Comment by vyasworks** [comment]
Overall looks good. Can attach a test result summary for future reference and documentation in this PR>

File: load/README.md (hunk 1 above)
Comment (arranfw) [reply load/README.md:12]: Good catch, thank you

**Review by fwpushan — DISMISSED** [review]

File: load/employerStaticFiles.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      maxVUs: rate,
+    },
+  },
+};
+
+export default function () {
+  const indexRes = http.get(`https://hcapemployers.${__ENV.OS_NAMESPACE_SUFFIX}.freshworks.club/`);
+
+  if (indexRes.status === 200) {
+    successCounter.add(1);
```
Comment (vyasworks) [inline load/employerStaticFiles.js:30]: we should add only one success counter for this entire function, since HTML + JS counts as one load.

Comment (arranfw) [reply load/employerStaticFiles.js:30]: Very good point, thanks!

**Review by vyasworks — APPROVED** [review]

**Pull request merged**